src/Keboola/S3Extractor/Application.php

```diff
@@ -3,71 +3,54 @@
 namespace Keboola\S3Extractor;
 
 use Aws\S3\Exception\S3Exception;
-use Monolog\Handler\HandlerInterface;
-use Monolog\Logger;
-use Symfony\Component\Config\Definition\Processor;
+use Keboola\Component\BaseComponent;
+use Keboola\Component\UserException;
 
-class Application
+class Application extends BaseComponent
 {
     /**
-     * @var array
+     * @throws UserException
      */
-    private $config;
+    protected function run(): void
+    {
+        /** @var Config $config */
+        $config = $this->getConfig();
 
-    /**
-     * @var array
-     */
-    private $parameters;
+        if (substr($config->getKey(), -1) === '/') {
```

Comment:
možná bych tuhle validaci hodil do `ConfigDefinition`? @zajca to parádně vyšvihl tady - https://github.com/keboola/input-mapping/pull/39/files#diff-a4296f364a5e819e99204f3ecb4884ceR57

Reply:
To se mi líbí, ale udělám to v samostatném PR https://github.com/keboola/aws-s3-extractor/issues/48